### DISCUSSION TITLE
Nautobot UI 2.0 Feedback ✨ 🎨

### DISCUSSION THREAD

**Discussion by bryanculver** [nautobot/nautobot#3872] [General]
Nautobot 2.0 is introducing an all-new UI, rebuilt from the ground up. Some comments may not necessarily fit within our existing issue template frameworks (suggested tweaks, questions, etc). Therefore this discussion can be used to provide feedback as comments and we can open them up as issues as needed.

This should not stop you from opening a bug or feature request you'd like to open for changes you'd like to see in the new UI.

**Comment by bryanculver** [comment]
For those that are curious, we will be building out our backlog and roadmap for new UI features and layouts we will be building in subsequent releases, such as:

**A Collapsable Prefix List View**
<img width="1920" alt="network-prefixes" src="https://github.com/nautobot/nautobot/assets/31187/84ddb537-90d0-4e4e-9c81-b9f4704dc6cc">

**Improved Rack Detail View**
<img width="1920" alt="inventory-rack" src="https://github.com/nautobot/nautobot/assets/31187/2f2258ba-faed-4b9b-b7f6-667768d49937">

**Advanced and Interactive Data Visualizations**
<img width="492" alt="onboarding_1" src="https://github.com/nautobot/nautobot/assets/31187/755ddd01-f410-46e3-8541-012fafe6831d">

Please feel free to share ideas you'd like to be built out, and if we already have plans in our design roadmap, we'll be sure to link to them.

**Comment by jeffkala** [comment]
On the home page, next.demo.nautobot.com are the Automations just mock ups? They show a number but clicking on them doesn't do anything.

Secondly, if you are clicking around on the tabs under the Automations section eventually the Changelog section refreshes and just says TODO

<img width="1228" alt="Screen Shot 2023-06-07 at 8 29 16 AM" src="https://github.com/nautobot/nautobot/assets/48843785/e64ba6e3-73b5-411c-afe1-adc3eba0a8aa">

**Reply by jeffkala** [reply]
Working on exact repro steps.

**Reply by jeffkala** [reply]
Actually don't have to click around at all, if you stay on the home page and monitor the network logs it keeps refreshing and getting 500 server errors and eventually this happens
Request URL: https://next.demo.nautobot.com/api/extras/object-changes/?depth=1&limit=10
Request Method: GET
Status Code: 500

**Reply by glennmatthews** [reply]
Thanks! Stack trace ends with:

```
  File "/source/nautobot/extras/api/serializers.py", line 695, in get_changed_object
    return return_nested_serializer_data_based_on_depth(self, depth, obj, obj.changed_object, "changed_object")
  File "/source/nautobot/core/api/utils.py", line 347, in return_nested_serializer_data_based_on_depth
    data = field_class(
  File "/source/nautobot/extras/api/serializers.py", line 121, in __init__
    self.fields.pop("dynamic_groups")

Exception Type: KeyError at /api/extras/object-changes/
Exception Value: 'dynamic_groups'
```

**Reply by glennmatthews** [reply]
To your first question, the `Schedule` and `Approvals` tabs are currently mockups and do need to be implemented. `Job History` should be working at this point though.

**Reply by glennmatthews** [reply]
Strange, I'm no longer able to replicate the KeyError issue, either on next.demo.nautobot.com or on my local instance. If you see it again, could you please file an issue with repro steps?

**Reply by jeffkala** [reply]
Will do, thanks @glennmatthews

**Reply by jeffkala** [reply]
I'm going to assume https://github.com/nautobot/nautobot/discussions/3872#discussioncomment-6134758 is related.

**Reply by jeffkala** [reply]
Validated to repro this, just make sure a extras | config context change-log is within the queryset of last 10 changelogs.

**Reply by glennmatthews** [reply]
Looks like it also requires hitting the API with `?depth=1` to trigger it. Thanks!

**Reply by bryanculver** [reply]
Addressed in https://github.com/nautobot/nautobot/pull/3896

**Comment by jeffkala** [comment]
I find the page navigation section in the new UI to be rather large and take up potentially valuable real estate on the page.  Would be curious if there are any thoughts to resize this section to make it smaller.
![Screen Shot 2023-06-07 at 11 32 45 AM](https://github.com/nautobot/nautobot/assets/48843785/59f43dc5-6df7-4457-8abf-e06b7a9806ea)

**Reply by bryanculver** [reply]
Moved to #3904

**Comment by lampwins** [comment]
The page loading widget forces the table on list pages down about 200px which causes a jarring transition when the content loads and it suddenly shifts upward. The loading widget is fine at the top of the page, just need to not shift the table.

**Comment by lampwins** [comment]
Need to reduce the cell padding in the tables, as each row takes up too much vertical space today.

**Reply by bryanculver** [reply]
Moved to #3905

**Comment by itdependsnetworks** [comment]
Feel free to edit this comment:

* You will get an `Invalid filters were specified` if you have a non default limit on the new ui and click on the "Return to Legacy UI" button. e.g. this `http://127.0.0.1:8080/dcim/racks/?limit=25&offset=0` in the legacy UI
* Unclear what Buttons do currently
   * Filter - Toast message
   * Actions - not working
   * Add - Prefix does nothing, other return to legacy ui 
   * Pencil - On individual object 
* Clicking the "configure" gear, can be too large and when you scroll down, you "leave" the viewable page. It suddenly just shows all white. 
* How does one see the following in the new UI? 
    * current version 
    * GraphQL link
    * API Docs Link
    * Switch to Dark UI
* `Rack Elevations` did not load, in chrome inspect i did not see an api call for it.
* `Interface Connections` did not load, but I captured the below stacktrace. The issue aside, in debug mode, not getting the stacktrace to the browser is concerning. 
* It is not clear that you can scroll to the right in a table, when the last column does not cut off in the middle
* "Provider Networks" default columns showed `Id` first, and Display on third column. Seems to be similar for all 5 models in "Inventory -> Circuits". Upon further review, this is the case for many models
* If 3 results you see `Showing 1 - 3 of 3`, if 0 results you see `Showing 1 - 50 of 0`
* `Content types` for at least `Location Types` show as `dcim.device` vs `dcim | device`
* `Content types` will go on forever, vs a truncated list with `...` on it
* A single `>` indicates to me, move one page, I would expect a `>>` to go to last page
* Linking to a page does not expand the nav panel to that page. This may be preferred, but want to be purposeful
* Jobs no longer has collapsible UI
* Jobs default columns has everything selected.
* `installed-plugins` doesn't seem to show the detailed view.
* "change log retention" information is missing from changelog view
* Color in "tags" no longer shows the color, but the hex number instead.
* Tags do not show up in Devices list view
* Tags do not show up in Devices detail view
* Racks do not have the `Space` and `Power` columns
* When opening home page, I get a 500 on API calls for `object-changes/` and `job-results`
* When adding via the "configure gear" the page scrolls off, so you cannot keep adding a bunch of columns.
* In Device detailed view, you can no longer link to all of the objects in the Location hierarchy. 
* A default `get_detail_view_config` on `ModelViewSetMixin` seems confusing to me. 


<img width="591" alt="image" src="https://github.com/nautobot/nautobot/assets/9260483/36917fc6-e66d-4d64-a86b-c43db83cac01">


Note that this is the Devices View but the data shows from the Platforms model, clicking on the view from another, still keeps the cache from either the prior Model view or even within the same view of the model, after a refresh (but not hard refresh/disable cache). 

<img width="1191" alt="image" src="https://github.com/nautobot/nautobot/assets/9260483/f7ebd098-f6d0-4f0b-ae10-54054b54dbb2">

**Comment by mzbroch** [comment]
- Navigating to https://next.demo.nautobot.com/plugins/installed-plugins/ Installed Plugins on Next demo leads to an empty page
- Objects have "Plugin Tab 1", "Plugin Tab 2" - not sure if this is due to `example_plugin`, but this plugin/tabs should not be in the demo.
- Schedule/and automation tabs indicate several objects, but nothing is displayed in the table in the https://next.demo.nautobot.com 
<img width="803" alt="Screenshot 2023-06-09 at 14 18 18" src="https://github.com/nautobot/nautobot/assets/16935122/bd82cf21-7abd-4c7a-b139-3b4701340f8b">

- Roles are no longer coloured
<img width="1083" alt="Screenshot 2023-06-09 at 14 22 38" src="https://github.com/nautobot/nautobot/assets/16935122/b77cfc86-ea7f-411d-908a-60af58f8a2b9">

- IP Address creation:
  - status field does not show anything
  - Prefix has to be typed statically - no auto-complete or auto-suggestion or drop-down is available

- In the list view , while looking at the results on the last page of the list (2) , Nautobot shows "next page" link

<img width="544" alt="Screenshot 2023-06-09 at 14 51 06" src="https://github.com/nautobot/nautobot/assets/16935122/2a70afcf-a387-45fd-a7e6-5a14bff20b80">

- device view is missing components represented through Device tabs in a device view

**Comment by jeffkala** [comment]
Loading changelog takes too long and results in a 500 error.
https://next.demo.nautobot.com/extras/object-changes/

Request URL: https://next.demo.nautobot.com/api/extras/object-changes/?depth=1
Request Method: GET
Status Code: 500 

It also caches the table size/rows from a previous page and nulls out the data which makes it appear that it loads "nothing".
![Screen Shot 2023-06-09 at 10 20 28 AM](https://github.com/nautobot/nautobot/assets/48843785/47cb8fe6-1d0c-404a-977f-4ce5d62fc128)

Even more odd, the full change-log is only 21 entries, so its not an overwhelming amount, the original page size I was using was 50. If setting it to 10 the page does load.

**Reply by jeffkala** [reply]
Also find it odd that an `ADD` action is showing up.  Don't think we would ever expect to manually create an Object Change.

**Reply by jeffkala** [reply]
Seems like all of these are related to a change object with content type of `extras | config context`, was playing around with sizing of the tables and going back and forth between legacy and new UI and its directly related to that change-object content type where the 
```
error
: 
"'dynamic_groups'"
exception
: 
"KeyError"
nautobot_version
: 
"2.0.0a3"
python_version
: 
"3.9.16"```

**Reply by jeffkala** [reply]
Think this is likely Addressed in https://github.com/nautobot/nautobot/pull/3896 as well

**Comment by travisjohnsonga** [comment]
Toggle dark theme missing in 2.0.1

**Comment by travisjohnsonga** [comment]
2.0.1  API has completely trashed my python scripts for syncing data to other sources.

Changing sites to locations
changing relationships from slugs to ids
the device role API is gone - also from web UI - I can't find device roles anywhere.

This is the worst API change I've ever seen - highly frustrated, and likely to go back to netbox

**Comment by hurleyp1** [comment]
Nautobot on mobile is a big value-add for my team to get site information on the go. With 2.0, the navigation menu takes up about 60% of the screen. It would be great to have it collapsible as it makes the current UI hard to work with:

![image](https://github.com/nautobot/nautobot/assets/58192379/ba536f13-309a-49de-b314-aa7f8d7748d2)

**Reply by bradh11** [reply]
I have some work ready to go for this.  Looking for some feedback on the look and feel.

https://github.com/nautobot/nautobot/assets/18483910/65bca7f4-3208-4397-83a9-328c8579b880

**Reply by hurleyp1** [reply]
Looks great, Brad! I think if you were to keep the ">>>" logo when collapsed, maybe that could be the element to select to re-expand the menu (thus removing the "->" element) but that is a very nitpicky opinion. I definitely look forward to its implementation

**Reply by bradh11** [reply]
For reference:  This will be in draft mode for a little bit, but early iteration can be seen here:

https://github.com/nautobot/nautobot/pull/5172
